File: DesafioHeroisFrontEnd/src/app/models/heroi.ts
import { Superpoder } from "./superpoder";

export class Heroi {
  id?: number;
  nome: string = '';
  nomeHeroi: string = '';
  dataNascimento: string = '';
  altura: number = 0;
  peso: number = 0;
  heroiSuperpoderes: Superpoder[] = [];

  constructor(init?: Partial<Heroi>) {
    Object.assign(this, init);
  }

  static map(data: any): Heroi {
    return new Heroi({
      ...data,
      dataNascimento: data.dataNascimento ? data.dataNascimento.split('T')[0] : ''
    });
  }


  static mapList(list: any[]): Heroi[] {
    return (list || []).map(item => Heroi.map(item));
  }


  static mapOutput(data: any): any {
    return {
      ...data,
      id: data.id || null,
      heroiSuperpoderes: data.powers.map((powerId: number) => ({
        powerId: powerId
      }))
    };
  }
}

File: HeroisAPI/Models/Heroi.cs
namespace HeroApp.Api.Models
{
    public class Heroi
    {
        public int Id { get; set; }
        public string Nome { get; set; } = string.Empty;
        public string NomeHeroi { get; set; } = string.Empty;
        public DateTime DataNascimento { get; set; }
        public double Altura { get; set; }
        public double Peso { get; set; }

        public List<HeroiSuperpoder> HeroisSuperpoderes { get; set; } = new();
    }


}
